omero/developers/Server/GraphsImpl.txt

```diff
@@ -0,0 +1,108 @@
+Java classes for model graph operations
+=======================================
+
+This description of the roles played by server-side Java classes assumes
+familiarity with the :doc:`ObjectGraphs` machinery of OMERO.server.
+
+
+Navigating the graph
+--------------------
+
+When OMERO.server starts up the :javadoc:`GraphPathBean
+<ome/services/graphs/GraphPathBean.html>` reflects upon the object model
+and collates information about classes, subclasses, properties and their
+value types. This is what :javadoc:`GraphPathReport
+<ome/services/graphs/GraphPathReport.html>` uses to generate the
+:doc:`../Model/EveryObject`.
+
+
+Traversing and acting
+---------------------
+
+OMERO.server's :javadoc:`GraphTraversal
+<ome/services/graphs/GraphTraversal.html>` is at the core of all graph
+operations,
+
+* querying the database to establish the model graph,
+  with the help of ``GraphPathBean``
+* applying the graph operation's policy rules and changing the graph
+  node states
+* acting on the model objects according to the final state of the graph.
+
+``GraphTraversal``'s :javadoc:`Processor
```

Comment:
Yes, this happens a few times across our docs: Sphinx guesses wrongly about quotes. I don't know of any way to stop it doing that.